### PR TITLE
Error if compiled nondeterministic backward called in deterministic mode

### PR DESCRIPTION
**Pull request by kurtamohler**
Part of #113707

cc @mruberry @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Review by ezyang — COMMENTED** [review]
Do you want this to error if it diverges either way, or is it perhaps OK if backwards is deterministic even if the user didn't ask for it?

**Review by ezyang — APPROVED** [review]

**Comment by kurtamohler** [comment]
> is it perhaps OK if backwards is deterministic even if the user didn't ask for it?

That's a good question. I can't think of any specific reason why the answer would be no. But I can think of a couple reasons why the answer might be yes: too many errors can be annoying and I feel like being more permissive is better if we don't see a reason to be restrictive. So I'll push an update shortly

File: torch/_functorch/_aot_autograd/schemas.py [hunk 1]
```diff
@@ -245,6 +245,8 @@ class ViewAndMutationMeta:
     # to reset the grad mode to its pre-graph value prior to calling aot_autograd.
     grad_enabled_mutation: Optional[bool] = None
 
+    deterministic: Optional[bool] = None
```
Comment (albanD) [inline torch/_functorch/_aot_autograd/schemas.py:248]: nit: add a comment explaining what this is and what it is used for.

**Review by albanD — APPROVED** [review]
SGTM!

**Pull request closed without merging**